### DISCUSSION TITLE
Invalid Logging Configuration when DynamoDB Replica creation

### DISCUSSION THREAD

**Discussion by racorrea** [aws/aws-cdk#28666] [Q&A] [answered]
I updated my cdk-lib to version 2.118.0 and when I try to create dynamodb replicas, it gives me this error:

`CREATE_FAILED        | AWS::StepFunctions::StateMachine | @aws-cdk--aws-dynamodb.ReplicaProvider/Provider/waiter-state-machine (Providerwaiterstatemachine5D4A9DF0) Resource handler returned message: "Invalid Logging Configuration: The CloudWatch Logs Resource Policy size was exceeded. We suggest prefixing your CloudWatch log group name with /aws/vendedlogs/states/. (Service: AWSStepFunctions; Status Code: 400; Error Code: InvalidLoggingConfiguration; Request ID: 9ca7e02a-07e3-4332-a5c3-1ca22ee511ae; Proxy: null)" (RequestToken: 7664d56a-1265-bfb7-996f-7e6d68e2bfa2, HandlerErrorCode: InvalidRequest)`

Has this happened to anyone else, do you know how to fix it? It should be noted that until version 2.110.0 - 2.115.0 this did not happen.
Thank you so much

**Comment by joseamoroso** [comment]
Hello @racorrea! I've seen this error in other services. The root cause is that you reached the [resource policy length limit](https://docs.aws.amazon.com/step-functions/latest/dg/bp-cwl.html). An option could be that you are using the default CloudWatch props for several StepFunctions. Then CDK re-uses the same policy, which leverages an extensive policy if you are deploying at an enterprise scale. You can identify the policy you're using with the AWS CLI:
```
aws logs describe-resource-policies 
```
 Then, there are two options here:

1. (Recommended) Output your logs to the `/aws/vendedlogs/` CloudWatch log group. This is a change from the `LogOptions` prop in the `StateMachine` class
2. Update the resource policy. This will require removing or updating a previous resource to allow a more comprehensive number of resources (using wildcards). However, this is not recommended as it will also remove granularity access.

**Comment by eliabrio** [comment, marked as answer]
Hi @racorrea, 

Had the same issue. 

Solved it by using the class `dynamodb.TableV2` instead of `dynamodb.Table` . The change in my case was rather minimal, from 

```python
        self.batch_table = dynamodb.Table(
            self, "GlobalTable",
            table_name=f"{stage_name}_global_table",
            partition_key=dynamodb.Attribute(
                name="global_table",
                type=dynamodb.AttributeType.STRING
            ),
            sort_key = dynamodb.Attribute(
                name="sort_key",
                type=dynamodb.AttributeType.STRING
            ),
            billing_mode=dynamodb.BillingMode.PAY_PER_REQUEST,
            removal_policy=RemovalPolicy.RETAIN if is_production else RemovalPolicy.DESTROY,
            replication_regions=["us-east-1", "us-east-2"]
        )
```

to

```python
        self.batch_table = dynamodb.TableV2(
            self, "GlobalTable",
            table_name=f"{stage_name}_global_table",
            partition_key=dynamodb.Attribute(
                name="global_table",
                type=dynamodb.AttributeType.STRING
            ),
            sort_key = dynamodb.Attribute(
                name="sort_key",
                type=dynamodb.AttributeType.STRING
            ),
            billing=dynamodb.Billing.on_demand(),
            removal_policy=RemovalPolicy.RETAIN if is_production else RemovalPolicy.DESTROY,
            replicas=["us-east-1", "us-east-2"]
        )
```
